src/codelabs/async-await/index.md

```diff
@@ -0,0 +1,346 @@
+ {{ site.alert.end }}
+
+
+## Async in Dart: What is a future?
+
+In Dart, a
+[Future]({{site.dart_api}}/Future-class.html)
+represents the result of an asynchronous operation. A future object has
+two states: uncompleted or completed.
+
+{{ site.alert.note }}
+'Uncompleted' is a special term in Dart referring to the state of a future 
+ before it has produced a result.
+{{ site.alert.end }}
+
+### Uncompleted
+
+When you invoke a function that returns a future, that function immediately 
+returns a future object in an uncompleted state. While in its uncompleted state,
+a future object is simply waiting for asynchronous operations to finish or to 
+throw an error. 
+
+### Completed
+
+There are two ways that a future object can complete:
+
+#### 1. Success
+
+If the asynchronous operations succeed, the future completes and returns a 
+result of type T. For example,  
+`Future<String>` is the type signature for a future object that produces a string
+result. If a future produces a result that isn't a usable value, then the
+future's type is `Future<void>`. 
+
+#### 2. Error
+
+If the asynchronous operations being performed by the 
+function throw an error, the future completes and returns an error. 
+
+To see this lifecycle at work consider what happens after invoking the 
+`getUserOrder` function:
+
+
+<iframe style="margin-top: 25px; margin-bottom: 25px" frameborder="no" height="450" src="https://dartpad.dartlang.org/experimental/embed-new.html?id=57e6085344cbd1719ed42b32f8ad1bce" width="100%"></iframe>
+
+
+After 1 second has passed, notice that the future is in an uncompleted state - you can't access the
+desired result yet:
+
+`     Your order is: Instance of '_Future<String>'`
+
+After 4 seconds have passed, the future completes and produces the desired
+result:
+
+`Your order is: 'Large Latte'`
+
+How do you access this result that completes 4 seconds later? In the next section
+you will learn about using  the `async` and `await` keywords to handle the results
+of asynchronous functions. 
+
+{{ site.alert.secondary }}
+Quick Review:
+
+* In Dart, a
+[Future]({{site.dart_api}}/Future-class.html)
+produces a result of type `<T>`. For example,
+[Future]({{site.dart_api}}/Future-class.html)`<String>`
+is the type signature for a future object that produces a string result. If a
+future produces a result that isn't a usable value, then the future's type is
+`Future<void>`.
+* A future can be in one of two states: uncompleted, or completed. 
+* A completed future can either (1) complete with a result or (2) complete with
+an error.
+* When you invoke a function that returns a future, two things happen:
+
+1. The function queues up work to be done and returns an uncompleted future.
+2. Later, when the operation is finished, the future completes with a value or
+with an error.
+{{ site.alert.end }}
+
+
+## Working with futures in Dart: async and await
+
+The `async` and `await` keywords provide a declarative way to: Define 
```

Comment:
The `async` and `await` keywords provide a declarative way to~~:~~ **d**efine 
asynchronous functions and access their results---*after they have completed.*

Reply:
The colon is a typo leftover from when this section used to be in list format. Thanks for the catch!